### DISCUSSION TITLE
Signed-in user favoriting/saving search

### DISCUSSION THREAD

**Discussion by willrlin** [bloom-housing/bloom#1719] [Ideas]
Discussion to talk about how signed in features might work where a user can favorite a listing or save a search or save their eligibility criteria from stepping through our front page questionnaire. @seanmalbert let us know if you have initial thoughts, we haven't started scoping out this work.

**Comment by seanmalbert** [comment]
Hey @willrlin ,

It seems like a simple structure with an interface like:
```
interface UserSettings {
  [key: string]: { label: string, value: string }[]
}
```
might work across the board, though I'm not sure about the eligibility criteria piece. For saved searches and listings it would be like:

```
{
  listings: [{ label: "Listing 1",  value: "some uuid" }, { label: "Listing 2",  value: "some uuid 2" } ],
  searches: [{ label: "Saved search label 1",  value: "query string" }],
}
```
But if we need value to be a type other than string, then I suppose that can be defined, but if we can keep it generic, it'd be easier to contribute upstream. So it might be easiest, at least for the first iteration, to use a jsonb column on user_accounts.

This settings column would be accessible under the user's profile, which is reachable via the frontend with AuthContext, and rather than calling the current user update method, it might be best to have a separate endpoint just for upserting a user's settings, so we can keep `update` clean.
